features/issues/components/filters-bar/filters-bar.tsx

```diff
@@ -0,0 +1,58 @@
+import { ButtonCTA, Select, Input, ButtonCTASize } from "@features/ui";
+import styles from "./filters-bar.module.scss";
+import { useEffect, useState } from "react";
+import { useFilters } from "@features/issues";
+import { Filters } from "@features/issues";
+
+export function FiltersBar() {
+  const { filters, handleFilters } = useFilters();
+
+  const [statusFilter, setStatusFilter] = useState(filters.status);
+  const [levelFilter, setLevelFilter] = useState(filters.level);
+  const [projectFilter, setProjectFilter] = useState(filters.project);
+
+  useEffect(() => {
+    handleFilters({
+      status: statusFilter,
+      level: levelFilter,
+      project: projectFilter,
+    });
+  }, [statusFilter, levelFilter, projectFilter]);
```

Comment:
You don't really need these state variables from what I can see. How this works now:

1. User updates a filter.
2. Filter state is updated.
3. Component is re-rendered.
4. useEffect is executed.
5. handleFilters is called and updates filters in URL
6. Component is re-rendered again.

If you simply remove the states and call `handleFilters` directly in the onChange handlers it would look like this:

1. User updates a filter.
2. handleFilters is called and updates filters in URL
3. Component is re-rendered again.

Same effect just fewer rendering cycles. Plus less complexity and potential of forgetting to sync state and URL.